### ISSUE TITLE
LIQE训练过程中loss不收敛，请问是什么问题呢

### ISSUE THREAD

**Issue by liuchenxin666** [chaofengc/IQA-PyTorch#229]
`
model = pyiqa.create_metric('liqe', device=device, pretrained=False)
model.to(device)

pred = model(img)
loss = loss_iqa(pred.squeeze(), mos.float().detach())
`

**Comment by liuchenxin666** [comment]
但是传入`as_loss=True`参数时，`InferenceModel`模型的`forward()`函数并没有传入mos值，`weight_reduce_loss()`函数如何计算的loss值。
同时，当`as_loss=True`时，`forward()`函数不再返回预测分数，那么我再计算测试集或者验证集上的SROCC，PLCC值时，是否无法继续进行呢？
因此，我不太理解`as_loss=True`参数的作用，我是否可以直接修改`InferenceModel`模型的`forward()`函数使其返回`output`呢？
```
    def forward(self, target, ref=None, **kwargs):
        device = self.device

        with torch.set_grad_enabled(self.as_loss):

            if self.metric_name == 'fid':
                output = self.net(target, ref, device=device, **kwargs)
            elif self.metric_name == 'inception_score':
                output = self.net(target, device=device, **kwargs)
            else:
                if not torch.is_tensor(target):
                    target = imread2tensor(target, rgb=True)
                    target = target.unsqueeze(0)
                    if self.metric_mode == 'FR':
                        assert ref is not None, 'Please specify reference image for Full Reference metric'
                        ref = imread2tensor(ref, rgb=True)
                        ref = ref.unsqueeze(0)
                        self.is_valid_input(ref)
                
                self.is_valid_input(target)

                if self.metric_mode == 'FR':
                    assert ref is not None, 'Please specify reference image for Full Reference metric'
                    output = self.net(target.to(device), ref.to(device), **kwargs)
                elif self.metric_mode == 'NR':
                    output = self.net(target.to(device), **kwargs)

        # if self.as_loss:
        #     if isinstance(output, tuple):
        #         output = output[0]
        #     return weight_reduce_loss(output, self.loss_weight, self.loss_reduction)
        # else:
        #     return output
        return output
```

**Comment by liuchenxin666** [comment]
此外，我理解的`weight_reduce_loss()`函数是在将`output`一个batch的预测分数求了个平均值？如果我理解的哪里有误，希望能够得到前辈指导，非常感谢！

**Comment by chaofengc** [comment]
这里的loss是指用训练好的模型作为一个另一个需要训练的模型，例如图像超分辨率模型的损失函数。如果你想要训练IQA模型本身，请参考本仓库训练部分的代码
